### PR TITLE
replace 'dummy' leadership key for activate latent weapon specials by real leadership abilities used like specials

### PR DESCRIPTION
**Pull request by newfrenchy83**
In 1.16, it is possible to use abilities like specials, like 'leadership' type.

### CODE REVIEW

**Comment by Dugy** [comment]
I'll have to postpone this merge until 1.16 becomes the main version and we can drop support for 1.14. Or 1.16 is out already?

**Comment by newfrenchy83** [comment]
> 
> 
> I'll have to postpone this merge until 1.16 becomes the main version and we can drop support for 1.14. Or 1.16 is out already?

1.16is yet in pre-string.

**Comment by Dugy** [comment]
I'll then postpone merging this until it's out. I wanted to make one final release for 1.14.

File: lua/stats.lua [hunk 1]
```diff
@@ -606,30 +594,22 @@ function wesnoth.update_stats(original)
...
-			elseif ability.id == "charge_leadership" then
-				remade.halo = "misc/charge-1.png:100,misc/charge-2.png:100,misc/charge-3.png:100,misc/charge-2.png:100"
-			elseif ability.id == "charge_leadership" then
-				remade.halo = "misc/charge-1.png:100,misc/charge-2.png:100,misc/charge-3.png:100,misc/charge-2.png:100"
-			end
+		elseif name == "berserk" and ability.id == "berserk_leadership" then
+			remade.halo = "misc/berserk-1.png:100,misc/berserk-2.png:100,misc/berserk-3.png:100,misc/berserk-2.png:100"
+		elseif name == "damage" and ability.id == "charge_leadership" then
+			remade.halo = "misc/charge-1.png:100,misc/charge-2.png:100,misc/charge-3.png:100,misc/charge-2.png:100"
+		elseif name = "poison" and ability.id == "poison_leadership" then
```
Comment (edwardspec) [inline lua/stats.lua:601]: typo: `name = "poison"` (should be `==`).

Comment (newfrenchy83) [reply lua/stats.lua]: fixed

**Comment by Dugy** [comment]
Merged this because 1.16 is now the main version.

Thanks for noticing this mess could be solved this cleanly in 1.16.

**Comment by Discontinuum** [comment]
Is it really intended that warlord's rule, drain leadership and berserk leadership work only in melee? Before it wasn't the case

**Comment by newfrenchy83** [comment]
> 
> 
> Is it really intended that warlord's rule, drain leadership and berserk leadership work only in melee? Before it wasn't the case

For drain leadership, the description said what drains applied to melle attack, and for the other, before the implementation in stats.lua in https://github.com/Dugy/Legend_of_the_Invincibles/commit/5d9fe5adeab97b93867b8e206fe3da44802487ba#diff-2f74a0e4bffe12df7717f21cc9af8250f4429bcf97a1c34b36cfec65f6294595 the melle ranged was condition for implement latent weapons specials but this condition don't remained.

**Comment by Discontinuum** [comment]
Do you want it to change @Dugy ? The latest lua implementation as uploaded in 1.14 had no ranges conditions except for the marksman leadership

**Comment by Dugy** [comment]
I think it made more sense to be melee only, I just forgot to keep it melee only.

**Pull request merged**